Lesson4/src/com/home/model/Fabric.java

```diff
@@ -61,7 +74,31 @@ public Model getModel(String requestedModel, MyReader myReader) throws IOExcepti
             }
             case "Address" -> {
                 System.out.println("-Please input String like: \"country city\"-");
-                return (Address) getModel("Address", myReader.someRead().get(0));
+                return getModel("Address", myReader.someRead().get(0));
+            }
+        }
+        return null;
+    }
+
+    public Model getModel(String requestedModel, PersonRegistry personRegistry, MyReader myReader) throws IOException {
+        switch (requestedModel) {
```

Comment:
switch with one scenario makes not too much sense for me.

I think if you are not going to modify this method to create anything rather than RecruitingOffice then you should rename method to getRecruitingOffice, get rid of requestedModel param and remove switch statement.

Same thing  for the method below

Reply:
ok, cleared